pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -222,8 +236,22 @@ func (c *Cluster) createDeviceSetPVC(existingPVCs map[string]*v1.PersistentVolum
 	if existingPVC != nil {
 		logger.Infof("OSD PVC %q already exists", existingPVC.Name)
 
+		desiredPvcSize, desiredOK := pvc.Spec.Resources.Requests[v1.ResourceStorage]
+		actualPvcSize, currentOK := existingPVC.Spec.Resources.Requests[v1.ResourceStorage]
+		if !desiredOK || !currentOK {
+			logger.Debugf("desired or current size are not specified for PVC %q", existingPVC.Name)
+			return existingPVC, nil
+		}
+
 		// Update the PVC in case the size changed
-		k8sutil.ExpandPVCIfRequired(c.clusterInfo.Context, c.context.Client, pvc, existingPVC)
+		resized := k8sutil.ExpandPVCIfRequired(c.clusterInfo.Context, c.context.Client, pvc, existingPVC)
+		if resized {
+			pvcResizeMap[existingPVC.Name] = pvcResize{
+				desiredSize: desiredPvcSize,
+				actualSize:  actualPvcSize,
+				resized:     false,
```

Comment:
It's confusing to see here that "if resized, then set resized to false". So maybe we can name the field differently?
```suggestion
				resizeConfirmed:     false,
```